### PR TITLE
Registry: Reject any non-enrolled public key from registering 

### PR DESCRIPTION
**Pull request by omerfirmak**
Part of #2289 

Will follow up with a Flash related PR.

### CODE REVIEW

File: doc/config.example.yaml [hunk 1]
```diff
@@ -269,3 +269,21 @@ event_handlers:
...
+  enabled: true
+
+  # DNS server configuration
+  dns:
+    enabled: true
+    address: 0.0.0.0
+    port: 53
+    authoritative:
+      - Zone1
+      - Zone2
```
Comment (Geod24) [inline doc/config.example.yaml]:
```suggestion
      - "validators.bosagora.io"
      - "flash.bosagora.io"
```

**Review by Geod24 — COMMENTED** [review]
Neat. Looks broadly LGTM, a few items stood out but I need to look deeper into it.
So I assume you don't have a use-case for a standalone registry ?
Also we probably need to change the UX a bit, and fold `dns` into `RegistryConfig`, but that can be done later. Since ports < 1024 are privileged under Linux, the nasty effect of having the registry on by default is that it needs to run as root (by default).

**Comment by omerfirmak** [comment]
>  So I assume you don't have a use-case for a standalone registry ?

Yeah, I see no reason why we should have that. But, we either need to fix Flash to not rely on registry or have a seperate registry  hat validates open channels for Flash.

IIUC, it works fine now, because Flash tests only use `LocalRest.Registry`, not `NameRegistry`

**Comment by Geod24** [comment]
Rebased and squashed related commit (otherwise the CI wouldn't pass on the first and second commit).

**Review by Geod24 — APPROVED** [review]

**Comment by Geod24** [comment]
@omerfirmak : Would it be complex to relax the restriction to allow not-yet-enrolled validators to register ?
This would allow a validator to ensure it's reachable from the second it becomes a validator.

**Comment by omerfirmak** [comment]
> 
> 
> @omerfirmak : Would it be complex to relax the restriction to allow not-yet-enrolled validators to register ?
> This would allow a validator to ensure it's reachable from the second it becomes a validator.

We could also check the enrollment pool. I'll take a look.

File: source/agora/node/Runner.d [hunk 2]
```diff
@@ -109,6 +116,30 @@ public Listeners runNode (Config config)
...
+    TCPListener dnstcplistener;
+    scope (exit)
+        if (config.registry.enabled && config.registry.dns.enabled)
+        {
+            // Here, we might need to interrupt the task to correctly shut down,
+            // however this throws an exception in the task that needs to be explicitly
+            // handled. And it doesn't help with the error messages printed by Vibe.d.
+            //dnstask.interrupt();
+            dnstcplistener.stopListening();
+        }
```
Comment (Geod24) [inline source/agora/node/Runner.d:128]: How did I miss this ? 🤦 
This closes the TCP listener as soon as this function exits. Instead, the listener should be stored in the `result`.

Comment (omerfirmak) [reply source/agora/node/Runner.d:128]: Shoot

**Pull request merged**